### PR TITLE
flamenco: make ALUT decode errors conformant + prevent OOB memory accesses

### PR DESCRIPTION
**Pull request by mjain-jump**
Also properly propagate errors from `fd_execute_txn_prepare_start`

### CODE REVIEW

File: src/flamenco/runtime/fd_executor.c [hunk 1]
```diff
@@ -384,23 +384,36 @@ fd_executor_setup_accessed_accounts_for_txn( fd_exec_txn_ctx_t * txn_ctx ) {
...
           .data = addr_lut_rec->const_data,
-          .dataend = &addr_lut_rec->const_data[56], // TODO macro const.
+          .dataend = &addr_lut_rec->const_data[FD_LOOKUP_TABLE_META_SIZE],
           .valloc  = fd_scratch_virtual(),
         };
-        if( fd_address_lookup_table_state_decode( &addr_lookup_table_state, &decode_ctx ) ) {
-          FD_LOG_ERR(("fd_address_lookup_table_state_decode failed"));
+
+        /* https://github.com/firedancer-io/agave/blob/368ea563c423b0a85cc317891187e15c9a321521/sdk/program/src/address_lookup_table/state.rs#L197-L198
+           https://github.com/firedancer-io/agave/blob/368ea563c423b0a85cc317891187e15c9a321521/sdk/program/src/address_lookup_table/state.rs#L205-L214 */
```
Comment (ibhatt-jumptrading) [inline src/flamenco/runtime/fd_executor.c:409]: why are there 2 comments here with differnet lines

File: src/flamenco/runtime/fd_executor.c [hunk 2]
```diff
@@ -384,23 +384,36 @@ fd_executor_setup_accessed_accounts_for_txn( fd_exec_txn_ctx_t * txn_ctx ) {
...
+        if( FD_UNLIKELY( fd_address_lookup_table_state_decode( &addr_lookup_table_state, &decode_ctx ) ) ) {
+          return FD_RUNTIME_TXN_ERR_INVALID_ADDRESS_LOOKUP_TABLE_DATA;
         }
+
+        /* https://github.com/firedancer-io/agave/blob/368ea563c423b0a85cc317891187e15c9a321521/sdk/program/src/address_lookup_table/state.rs#L200-L203 */
         if( addr_lookup_table_state.discriminant != fd_address_lookup_table_state_enum_lookup_table ) {
```
Comment (ibhatt-jumptrading) [inline src/flamenco/runtime/fd_executor.c:415]: nit can you make this fd_unlikely

File: src/flamenco/runtime/fd_executor.c (hunk 1 above)
Comment (mjain-jump) [reply src/flamenco/runtime/fd_executor.c:409]: Because the logic is split up in a weird if condition

File: src/flamenco/runtime/fd_runtime.c [hunk 3]
```diff
@@ -780,8 +780,9 @@ fd_runtime_prepare_txns_start( fd_exec_slot_ctx_t *         slot_ctx,
 
     int res = fd_execute_txn_prepare_start( slot_ctx, txn_ctx, txn_descriptor, &raw_txn );
     if( res ) {
-      txn->flags = 0U;
-      result     = res;
+      task_info[txn_idx].exec_res = res;
```
Comment (ibhatt-jumptrading) [inline src/flamenco/runtime/fd_runtime.c:783]: why dont we just set task_info[txn_idx].exec_res and return it

**Review by ibhatt-jumptrading — APPROVED** [review]

**Pull request merged**